spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -120,15 +122,16 @@ public ExecutorProvider subscriberExecutorProvider() {
 	}
 
 	@Bean
+	@ConditionalOnBean(type = "com.fasterxml.jackson.databind.ObjectMapper")
```

Comment:
@artembilan What am I doing wrong here? It doesn't seem to be working. I get 
`No qualifying bean of type 'org.springframework.cloud.gcp.pubsub.support.converter.PubSubMessageConverter' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}`.

Reply:
??? You don't have an `ObjectMapper` bean in your application context, therefore this `JacksonPubSubMessageConverter` is not created as well. And your consumer `PubSubTemplate` fails to be inject because there is no any beans on the matter.

However I see you have there already `SimplePubSubMessageConverter...

So, what, where and how fails.

Maybe you can share more stack trace to determine?

Thanks

Reply:
Why do you use a type name instead of class?
It doesn't bring me too much sense since you are going to use this type explicitly in the bean method argument.